BanyanImages/src/locations.jl

```diff
@@ -0,0 +1,161 @@
+function RemotePNGSource(remotepath; shuffled=false, source_invalid = false, sample_invalid = false, invalidate_source = false, invalidate_sample = false)::Location
+    RemoteSource(
+        remotepath,
+        shuffled=shuffled,
+        source_invalid = source_invalid,
+        sample_invalid,
+        invalidate_source = invalidate_source,
+        invalidate_sample = invalidate_sample
+    ) do remotepath, remote_source, remote_sample, shuffled
+        # Detect whether this is a png file
+        png_ending = ".png"
+        isa_png = occursin(png_ending, remotepath)
+
+        if !isa_png
+            error("Expected PNG dataset for $remotepath")
+        end
+
+        p = download_remote_path(remotepath)
+
+        # Initialize parameters if location is already cached
+        files = isnothing(remote_source) ? [] : remote_source.files
+        ndims = isnothing(datandims) ? 0 : remote_source.ndims
+        nbytes = isnothing(remote_source) ? 0 : remote_source.nbytes
+        dataeltype = isnothing(remote_source) ? nothing : remote_source.dataeltype
+
+        # Determine if this is a directory
+        p_isfile = isfile(p)
+        newp_if_isdir = endswith(string(p), "/") ? p : (p * "/")
+        p_isdir = !p_isfile && isdir(newp_if_isdir)
+        if p_isdir
+            p = newp_if_isdir
+        end
+
+        # Get files to read
+        files_to_read_from = if p_isdir
+            [joinpath(remotepath, filep) for filep in Random.shuffle(readdir(p))]
+        elseif p_isfile
+            [p]
+        else
+            []
+        end
+
+        # Initialize sample
+        randomsample = nothing
+
+        if isnothing(remote_sample)
+
+            # TODO: How to get this?
+            samplesize = 2
```

Comment:
You can determine the size of the sample by dividing the number of files by the `get_job().sample_rate` (I think that's how you get the job's sample rate.

Reply:
Look at `ReadTableSource` but you might also have to lowerbound it to the the exact `Banyan.get_max_exact_sample_length()`.